src/components/calcite-button/calcite-button.tsx

```diff
@@ -131,7 +131,12 @@ export class CalciteButton {
     if (this.iconposition === "start") {
       return (
         <Host dir={dir} hastext={this.hastext}>
-          <Tag {...attributes} role={role} disabled={this.disabled}>
+          <Tag
+            tabindex="0"
+            {...attributes}
```

Comment:
The tabIndex and role are unnecessary if the Tag is a button. Maybe that logic could be handled in the `attributes`?

Reply:
Is there a reason to use an anchor tag if the tag won't ever have a href? maybe a span would be better since the button role will be there?

Reply:
@driskull good feedback....

Hm, yeah I guess role and tabindex are inferred by default when the tag is `<button>` or `<a>` already aren't they 😛 ?

Only reason for using `<a>` in the case of "appearance='inline'" and NO href was that it also got assigned a role="link" in the existing logic, I can just assign tabindex="0" and role="link" to a `<span>` explicitly for that use case.